### PR TITLE
{REST} `az rest`: Update help example for passing JSON 

### PR DESCRIPTION
**Pull request by jiasli**
**Related command**
`az rest`

**Description**
Escaping double quotes in JSON is tedious and error-prone, this update updates help message for passing JSON to show more examples.

### CODE REVIEW

File: doc/quoting-issues-with-powershell.md [hunk 1]
```diff
@@ -1,41 +1,5 @@
 # Quoting issues with PowerShell
 
-**These issues have been fixed in Azure CLI 2.40.0 and PowerShell 7.3. Please update Azure CLI and PowerShell to the latest versions to avoid these issues.**
```
Comment (jiasli) [inline doc/quoting-issues-with-powershell.md:3]: We have to roll back the change due to the reverting of `ps1` entry script (https://github.com/Azure/azure-cli/pull/24015).

File: src/azure-cli/azure/cli/command_modules/util/_help.py [hunk 2]
```diff
@@ -16,15 +16,26 @@
...
 examples:
   - name: Get Audit log through Microsoft Graph
     text: >
         az rest --method get --url https://graph.microsoft.com/beta/auditLogs/directoryAudits
   - name: Update a Azure Active Directory Graph User's display name
-    text: >
-        az rest --method patch --url "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body "{\\"displayName\\": \\"jondoe2\\"}"
+    text: |
+        (Bash or CMD)
+        az rest --method patch --url "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body "{\\"displayName\\": \\"johndoe2\\"}"
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/_help.py:32]: In `\\`, the first `\` is for YAML to escape the second `\`. The rendered result of `--help` is 

```
"{\"displayName\": \"johndoe2\"}"
```

File: doc/quoting-issues-with-powershell.md (hunk 1 above)
Comment (bebound) [reply doc/quoting-issues-with-powershell.md:3]: Should we tell user `azps` is kept?

Comment (jiasli) [reply doc/quoting-issues-with-powershell.md:3]: `azps` is too buggy at the comment. Let's do that in separate PRs.

**Review by bebound — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**